### ISSUE TITLE
Free memory after rewinds are disabled in bodyWrapper

### ISSUE THREAD

**Issue by gavv** [gavv/httpexpect#381]
When bodyWrapper.DisableRewinds is called, we forbid further rewinds. From that point, user can only read response once until EOF or error.

However, if we have response in memory, we do not free this memory even though it's not needed anymore.

To free it, it's enough to do:

```go
bw.memReader = nil // or maybe bytes.NewReader(nil)
bw.memBytes = nil
```

The tricky part is that DisableRewinds can be called while we're in the middle of reading response from memory. In this case, DisableRewinds can't free memory right now; instead, it should be freed only when read encounters EOF.

I think no new tests are needed for this feature, it would be enough to modify existing tests, specifically TestBodyWrapper_DisableRewinds, which already has checks for allocated memory in various cases.

**Comment by gpjservais** [comment]
Hey @gavv, could I work on this one?

**Comment by gavv** [comment]
Sure!

**Comment by gavv** [comment]
Landed via 6ff151ac30802786a1263874225a450f6f41df24, thanks!
